File: backend/app/schemas/role.py
from pydantic import BaseModel, ConfigDict


class RoleBase(BaseModel):
    nombre: str
    descripcion: str | None = None


class RoleCreate(RoleBase):
    pass


class RoleUpdate(BaseModel):
    nombre: str | None = None
    descripcion: str | None = None


class RoleResponse(RoleBase):
    id_rol: int

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/role.ts
export type Role = {
  id_rol: number;
  nombre: string;
  descripcion: string | null;
};

export type CreateRolePayload = {
  nombre: string;
  descripcion?: string | null;
};

export type UpdateRolePayload = {
  nombre?: string | null;
  descripcion?: string | null;
};
